python/pyarrow/tests/test_flight.py

```diff
@@ -570,3 +570,17 @@ def test_tls_do_get():
             server_location, tls_root_certs=certs["root_cert"])
         data = client.do_get(flight.Ticket(b'ints')).read_all()
         assert data.equals(table)
+
+
+def test_tls_override_hostname():
+    """Check that incorrectly overriding the hostname fails."""
+    certs = example_tls_certs()
+
+    with flight_server(
+            ConstantFlightServer, tls_certificates=certs["certificates"]
```

Comment:
Ok... you have to pass the `connect_args` as above, otherwise `flight_server` may be stuck in a loop trying to reconnect.

Reply:
>.> Right, thanks.